packages/core/mesh/messaging/src/signal-client/signal-client-monitor.ts

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { trace } from '@dxos/tracing';
+
+import type { Message } from '../signal-methods';
+
+export class SignalClientMonitor {
```

Comment:
I (think) I understand the difference between the messenger and the signal-client, but I wonder for the sake of observability whether it'd be beneficial to hide that complexity. With a lot of these metrics it'll be difficult to discern what is due to network connectivity/system load vs system or software issues, I think the highest fidelity might be in in:
P0
* number of sent messages (overall amount of traffic)
* number of rate limited messages

P1
* number of retransmissions 
* number of connection aborted due to failure
* number of failed messages

in general, less is better

Reply:
>  but I wonder for the sake of observability whether it'd be beneficial to hide that complexity

you mean changing metric names (`mesh.signal.signal-client.received-total` -> `mesh.signal.received-total`)?
I'm ok with having a small number of core class-agnostic metrics like sent message count, call duration distribution etc. 
but I'd also keep class level metrics. the idea here is to use counters for understanding how often (if at all) certain code paths are taken. revisiting the PR after two weeks I think "recordLongGc" might be a bad example. if this starts happening, this will likely be a rare event (easy to miss with time series counter) and I'll probably want to examine at least some of the sessions in detail.
but there might be some code branches expected to be taken, but only rarely. logging warning (medium-priority sentry issues) is an overkill, but I want to have a cheap way for getting a signal when the branch starts being taken often (with case-based definition of often). I used to do this with logs, but a counter sounds like a better way in our env